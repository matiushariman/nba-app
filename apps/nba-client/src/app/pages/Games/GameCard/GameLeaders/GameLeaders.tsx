import GameLeaderDetails from './GameLeaderDetails';

import type { GameLeadersProps } from './GameLeaders.types';

const TABLE_HEADER = ['Pts', 'Reb', 'Ast'];

const GameLeaders = ({ gameLeaders }: GameLeadersProps) => {
  const GAME_LEADERS = [gameLeaders.awayLeaders, gameLeaders.homeLeaders];

  return (
    <div className="pt-4 pb-2 pl-4 pr-4 grid gap-2">
      <p className="font-bold text-sm uppercase">Game Leaders</p>
      <div>
        <table className="table-auto w-full">
          <thead className="border-b">
            <tr>
              <th className="w-[60%]" />
              {TABLE_HEADER.map((header, i) => (
                <th
                  key={`gameLeaderTh-${i}`}
                  className="text-xs uppercase text-right"
                >
                  {header}
                </th>
              ))}
            </tr>
          </thead>
          <tbody>
            {GAME_LEADERS.map((gameLeader) => (
              <tr key={gameLeader.personId}>
                <td>
                  <GameLeaderDetails
                    name={gameLeader.name}
                    jerseyNum={gameLeader.jerseyNum}
                    teamTricode={gameLeader.teamTricode}
                    position={gameLeader.position}
                    personId={gameLeader.personId}
                  />
                </td>
                <td className="text-sm text-right">{gameLeader.points}</td>
                <td className="text-sm text-right">{gameLeader.rebounds}</td>
                <td className="text-sm text-right">{gameLeader.assists}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default GameLeaders;
