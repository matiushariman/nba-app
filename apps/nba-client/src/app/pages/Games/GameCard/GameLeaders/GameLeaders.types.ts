import type { Game } from '@nba-app/types-nba';

export type GameLeadersProps = Pick<Game, 'gameLeaders'>;
