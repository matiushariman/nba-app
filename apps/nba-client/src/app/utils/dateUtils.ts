import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
