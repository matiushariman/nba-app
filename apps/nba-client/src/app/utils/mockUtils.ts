export const shouldUseMock = process.env['NODE_ENV'] === 'development';
