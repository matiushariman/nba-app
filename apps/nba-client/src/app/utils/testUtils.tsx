import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
