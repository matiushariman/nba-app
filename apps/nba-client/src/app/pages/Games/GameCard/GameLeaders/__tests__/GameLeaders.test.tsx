import GameLeaders from '..';

import { render } from '../../../../../utils/testUtils';

describe('pages/Games/GameCard/GameLeaders', () => {
  const gameLeaders = {
    homeLeaders: {
      personId: 202691,
      name: 'Klay Thompson',
      jerseyNum: '11',
      position: 'G',
      teamTricode: 'GSW',
      playerSlug: 'klay-thompson',
      points: 17,
      rebounds: 1,
      assists: 1,
    },
    awayLeaders: {
      personId: 1630188,
      name: 'Jalen Smith',
      jerseyNum: '25',
      position: 'FC',
      teamTricode: 'IND',
      playerSlug: 'jalen-smith',
      points: 9,
      rebounds: 6,
      assists: 1,
    },
  };

  it('should render GameLeaderDetails that matches snapshot', () => {
    const { baseElement } = render(<GameLeaders gameLeaders={gameLeaders} />);

    expect(baseElement).toMatchSnapshot();
  });
});
