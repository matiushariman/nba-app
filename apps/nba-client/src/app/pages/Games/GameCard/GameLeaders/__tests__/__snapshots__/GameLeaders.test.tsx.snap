// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Games/GameCard/GameLeaders should render GameLeaderDetails that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="pt-4 pb-2 pl-4 pr-4 grid gap-2"
    >
      <p
        class="font-bold text-sm uppercase"
      >
        Game Leaders
      </p>
      <div>
        <table
          class="table-auto w-full"
        >
          <thead
            class="border-b"
          >
            <tr>
              <th
                class="w-[60%]"
              />
              <th
                class="text-xs uppercase text-right"
              >
                Pts
              </th>
              <th
                class="text-xs uppercase text-right"
              >
                Reb
              </th>
              <th
                class="text-xs uppercase text-right"
              >
                Ast
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <div
                  class="flex items-center pt-1 pb-1"
                >
                  <div
                    class="mr-2 w-[24px] h-[24px]"
                  >
                    <img
                      alt="Jalen Smith headshot"
                      class="mr-2 bg-gray-300 shadow-sm rounded-full w-full h-full object-cover"
                      loading="lazy"
                      src="https://cdn.nba.com/headshots/nba/latest/260x190/1630188.png"
                    />
                  </div>
                  <div
                    class="grid"
                  >
                    <p
                      class="text-sm"
                    >
                      Jalen Smith
                    </p>
                    <div
                      class="grid grid-cols-3 divide-x gap-2"
                    >
                      <p
                        class="text-xs"
                      >
                        IND
                      </p>
                      <p
                        class="text-xs"
                      >
                        #25
                      </p>
                      <p
                        class="text-xs"
                      >
                        FC
                      </p>
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="text-sm text-right"
              >
                9
              </td>
              <td
                class="text-sm text-right"
              >
                6
              </td>
              <td
                class="text-sm text-right"
              >
                1
              </td>
            </tr>
            <tr>
              <td>
                <div
                  class="flex items-center pt-1 pb-1"
                >
                  <div
                    class="mr-2 w-[24px] h-[24px]"
                  >
                    <img
                      alt="Klay Thompson headshot"
                      class="mr-2 bg-gray-300 shadow-sm rounded-full w-full h-full object-cover"
                      loading="lazy"
                      src="https://cdn.nba.com/headshots/nba/latest/260x190/202691.png"
                    />
                  </div>
                  <div
                    class="grid"
                  >
                    <p
                      class="text-sm"
                    >
                      Klay Thompson
                    </p>
                    <div
                      class="grid grid-cols-3 divide-x gap-2"
                    >
                      <p
                        class="text-xs"
                      >
                        GSW
                      </p>
                      <p
                        class="text-xs"
                      >
                        #11
                      </p>
                      <p
                        class="text-xs"
                      >
                        G
                      </p>
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="text-sm text-right"
              >
                17
              </td>
              <td
                class="text-sm text-right"
              >
                1
              </td>
              <td
                class="text-sm text-right"
              >
                1
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</body>
`;
